---
output: 
  html_document:
    code_download: false
    toc: true                  # table of content true
    toc_depth: 5               # up to five depths of headings (specified by #, ## and ###)
    toc_float: true
    number_sections: true      # if you want number sections at each table header
    theme: united              # many options for theme.
    highlight: tango           # specifies the syntax highlighting style
    css: 'style.css'
params:
  ptable_gene_simple: "pivot_gene_simple.tsv"
  ptable_gene_complete: "pivot_gene_complete.tsv"
  ptable_variant: "pivot_variant.tsv"
  pplot_tiers: "tiers.png"
title: "`r paste0('PCGR-nf Report' , '') `"
author: ""
date: ""
---

<img src="logo.png" width=100 style="position:absolute;top:4px;bottom:4px;right:4px;" />

```{r include = FALSE}
knitr::opts_chunk$set(echo=FALSE)
```

```{r echo = FALSE, warning = FALSE}
source("DTable.R")
source("sanitise.R")
suppressPackageStartupMessages({
library(magrittr)
library(DT)
library(data.table)
library(htmltools)
library(stringr)
library(dplyr)
library(kableExtra)
   })
```

This is the summary report for for variant interpretation for precision cancer medicine using [Personal Cancer Genome Reporter (PCGR)](https://github.com/sigven/pcgr). Here you can see the summary information for somatic variants annotated, as well as by-gene and by-variant level information.

## Summary Plots

```{r, warning=FALSE}
figure_number  <- "Figure 1: "
figure_caption <- "PCGR Tiers - Number of unique variants per tier"
knitr::include_graphics(params$pplot_tiers, )
htmltools::tags$figcaption( style = 'caption-side: bottom; text-align: center; font-size: 85%%; color: #71879d',
                            htmltools::em(figure_number),
                            htmltools::em(figure_caption))
```

## Summary Tables

### By Gene

#### Gene

```{r, warning=FALSE}
table_genes <- as.data.frame(data.table::fread(params$ptable_gene_simple, sep="\t"))
DTable(table_genes)
```

#### Gene, Variant Class and Consequence

```{r, warning=FALSE}
table_genes <- as.data.frame(data.table::fread(params$ptable_gene_complete, sep="\t"))
DTable(table_genes)
```

### By Variant

```{r, warning=FALSE}
table_variants <- as.data.frame(data.table::fread(params$ptable_variant, sep="\t"))
DTable(table_variants)
```
